sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -142,6 +142,12 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
         joins.SortMergeOuterJoin(
           leftKeys, rightKeys, RightOuter, condition, planLater(left), planLater(right)) :: Nil
 
+      case ExtractEquiJoinKeys(FullOuter, leftKeys, rightKeys, condition, left, right)
```

Comment:
I think you can coalesce this and the 2 cases above into

case ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, condition, left, right) if ...
RowOrdering.isOrderable(leftKeys) is enough, we don't need to check right since the leftKeys and rightKeys should have the same orderable property.